database/sql/V033__new_table_pre_interactions.sql

```diff
@@ -0,0 +1,10 @@
+-- Creates new table for pre_interactions of orders
+
+CREATE TABLE pre_interactions (
```

Comment:
Does Postgres guarantee ordering? i.e. if we insert pre-interactions in a certain order, is querying them guaranteed to return in the same order? This isn't obvious to me (maybe its just my lack of Postres knowledge?) but is important - especially if an order needs more than one interaction executed in a specific order.

If it is not guaranteed, then we would need an index.

Also, is it worth having a:
```sql
CREATE TYPE InteractionKind AS ENUM ('pre' /* eventually: 'intra', 'post' */);

CREATE TABLE interactions (
    order_uid bytea NOT NULL,
    index int NOT NULL,
    kind InteractionKind NOT NULL,
    target_to bytea NOT NULL,
    value numeric(78,0) NOT NULL,
    data bytea NOT NULL
);
```

I may also be over-designing this ATM, but it would be nice to keep things open to allow for new interactions "kinds" to be added in the future. Maybe just renaming the table to `interactions` and having them implicitly being "pre"-interactions for now (and adding an interaction kind column later) might be enough. WDYT?

Reply:
I don't think an ordering is guaranteed. If we think that pre-interactions should be able to rely on an ordering then we indeed need a way to enforce that.

Reply:
I don't think POstgres guarantees ordering. 
You are right that I assumed that pre_interactions can be executed in arbitrary order. E.g. for "ethflow-pre_interactions" or "approval via signature" the order should not make a difference.


>Maybe just renaming the table to interactions and having them implicitly being "pre"-interactions for now (and adding an interaction kind column later) might be enough.

This sounds good. I think adding the kind can be done straight forward. I am in favour of it, as I can not think of a use case for intra and post interactions right now.

I will also add an index. Good catch.

Reply:
> I will also add an index.

This is a design decisions. Should pre interactions be able to depend on an ordering?

Reply:
>Should pre-interactions be able to depend on an ordering?

I think currently it's not needed. But looking forward, it might be. Hence to be safe, it makes sense to introduce it already?

Reply:
I'm not sure. It's hard for me to imagine how they will be used. If we want an ordering then I guess we can add an integer column to specify it.

Reply:
I implemented it now in the PR. :) I guess it does not hurt to have it in already.